Modules/Office365DSC/DSCResources/MSFT_SPOHomeSite/MSFT_SPOHomeSite.psm1

```diff
@@ -0,0 +1,197 @@
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.String]
+        $Url,
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Setting configuration for home site '$Url'"
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+        -Platform SharePointOnline
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+        -Platform PnP
+
+    $currentValues = Get-TargetResource @PSBoundParameters
+
+    if ($Ensure -eq "Present" -and $currentValues.Ensure -eq "Absent" `
+            -or $Ensure -eq "Present" -and $currentValues.Ensure -eq "Present")
+    {
+        try
+        {
+            Write-Verbose -Message "Setting home site collection $Url"
+            Get-SPOSite $Url
+        }
+        catch
+        {
+            $Message = "The specified Site Collection $($Url) for SPOHomeSite doesn't exist."
+            New-Office365DSCLogEntry -Error $_ -Message $Message
+            throw $Message
+        }
+
+        Write-Verbose -Message "Configuring site collection as Home Site"
+        Set-PnPHomeSite -Url $Url
+    }
+    else
+    {
+        # Remove home site
+        Remove-PnPHomeSite -Force
+    }
+}
+
+function Test-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("Yes")]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.String]
+        $Url,
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Testing configuration for home site collection"
+
+    $CurrentValues = Get-TargetResource @PSBoundParameters
+
+    Write-Verbose -Message "Current Values: $(Convert-O365DscHashtableToString -Hashtable $CurrentValues)"
+    Write-Verbose -Message "Target Values: $(Convert-O365DscHashtableToString -Hashtable $PSBoundParameters)"
+
+    $valuesToCheck = @("Ensure")
+    if ($PSBoundParameters.ContainsKey("Url"))
+    {
+        $valuesToCheck += "Url"
+    }
+
+    $TestResult = Test-Office365DSCParameterState -CurrentValues $CurrentValues `
+        -Source $($MyInvocation.MyCommand.Source) `
+        -DesiredValues $PSBoundParameters `
+        -ValuesToCheck $valuesToCheck
+
+    Write-Verbose -Message "Test-TargetResource returned $TestResult"
+
+    return $TestResult
+}
+
+function Export-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.String])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("Yes")]
+        [String]
+        $IsSingleInstance,
```

Comment:
Export-TargetResource should only ever accept GlobalAdminAccount as a parameter and occasionally MaxProcesses for multi-threaded extracts. Refer to another resources to look at how we are doing it.